### PR TITLE
RCT/JX/Rule5-18

### PR DESCRIPTION
**Pull request by Jiarongx-Xie**
This rule has been tested.

### CODE REVIEW

**Comment by yunjoonjung-PNNL** [comment]
Jiarong, could you switch the master branch to develop? Thanks!

**Review by yunjoonjung-PNNL — APPROVED** [review]
Great work Jiarong! Approved.

**Review by weilixu — APPROVED** [review]
Great work @Jiarongx-Xie, you followed the coding styles and have done excellent work to finish this task.
I have two comments to your PR. I approve the PR but please address the two comments.

**Comment by Jiarongx-Xie** [comment]
> Great work @Jiarongx-Xie, you followed the coding styles and have done excellent work to finish this task. I have two comments to your PR. I approve the PR but please address the two comments.

Thanks Weili! I appreciate all your guidance and comments!

File: rct229/rulesets/ashrae9012019/section5/section5rule18.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+            return has_existing_or_altered_space_b
+
+        def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+            has_existing_or_altered_space_b = calc_vals[
+                "has_existing_or_altered_space_b"
+            ]
+
+            return (
+                f"Part or all of zones listed below is existing or altered. The baseline vertical fenestration "
```
[the remark this answers is not in the document]
Comment (Jiarongx-Xie) [reply rct229/rulesets/ashrae9012019/section5/section5rule18.py:78]: @weilixu then should we remove `  existing_or_altered_space_list_b = calc_vals[
                "existing_or_altered_space_list_b"
            ]`

[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/rulesets/ashrae9012019/section5/section5rule18.py]: @Jiarongx-Xie you can still include it in the message. The `...` does not mean no message but keep the same as the original one.

[the remark this answers is not in the document]
Comment (Jiarongx-Xie) [reply rct229/rulesets/ashrae9012019/section5/section5rule18.py]: @weilixu I figured it out and that's what I did in the latest commit. Thank you for the clarification!

**Review by weilixu — APPROVED** [review]

**Pull request merged**